### ISSUE TITLE
Can't configure custom user and group backends

### ISSUE THREAD

**Issue by swizzly** [Icinga/icingaweb2#2840]
Hi
Having a Custom-Group-Backend without GUI causes a problem in Icingaweb2 (see below).
Maybe Icingaweb2 should make a small message indicating that the Backend has no GUI, or cannot show the list of groups.
Thanks

Message that currently appears:
Invalid backend type "mybackendname" provided

```
#0 /usr/share/icingaweb2/application/forms/Config/UserGroup/UserGroupBackendForm.php(179): Icinga\Forms\Config\UserGroup\UserGroupBackendForm->getBackendForm('mybackendname')
#1 /usr/share/php/Icinga/Web/Form.php(772): Icinga\Forms\Config\UserGroup\UserGroupBackendForm->createElements(Array)
#2 /usr/share/php/Icinga/Web/Form.php(1097): Icinga\Web\Form->create(Array)
#3 /usr/share/icingaweb2/application/forms/Config/UserGroup/UserGroupBackendForm.php(191): Icinga\Web\Form->populate(Array)
#4 /usr/share/php/Icinga/Web/Form.php(1183): Icinga\Forms\Config\UserGroup\UserGroupBackendForm->onRequest()
#5 /usr/share/icingaweb2/application/controllers/UsergroupbackendController.php(92): Icinga\Web\Form->handleRequest()
#6 /usr/share/icingaweb2/library/vendor/Zend/Controller/Action.php(507): Icinga\Controllers\UsergroupbackendController->editAction()
#7 /usr/share/php/Icinga/Web/Controller/Dispatcher.php(76): Zend_Controller_Action->dispatch('editAction')
#8 /usr/share/icingaweb2/library/vendor/Zend/Controller/Front.php(937): Icinga\Web\Controller\Dispatcher->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))
#9 /usr/share/php/Icinga/Application/Web.php(389): Zend_Controller_Front->dispatch(Object(Icinga\Web\Request), Object(Icinga\Web\Response))
#10 /usr/share/php/Icinga/Application/webrouter.php(109): Icinga\Application\Web->dispatch()
#11 /usr/share/icingaweb2/public/index.php(4): require_once('/usr/share/php/...')
#12 {main}
```

**Comment by lippserd** [comment]
Hi,

Thanks for the report. Unfortunately we did not think about how to configure custom group and user backends at all. So even if you would provide a form, it would fail. I change the title of the issue accordingly. We'll have a look at this.

Best regards,
Eric

**Comment by nilmerg** [comment]
We did think about it. Though, we deliberately ignored it since it was unlikely that anyone encounters this...

**Comment by nilmerg** [comment]
Resolved by #4365 and #4393
